canvas_app.py

```diff
@@ -183,11 +184,14 @@ async def delete_course(course_id: int):
         return JSONResponse(status_code=404, content={"message": "Courses not found"})
     courses = conf.get_conf()["courses"]
     all_courses = []
-    for course in courses:
-        if course["course_id"] != course_id:
-            all_courses.append(course)
-    conf.set_key_value("courses", all_courses)
-    return JSONResponse(status_code=200, content={"message": "success"})
+    if not isinstance(courses,Iterable):
```

Comment:
What does `not instance` do?

Reply:
`not Isinstance` is designed for type checck. If coursse is not a iterable variable (actually, I think the type of courses should be a list of dictionary), then  the code should raise an error message.